src/coreclr/src/jit/morph.cpp

```diff
@@ -18395,19 +18391,21 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
         assert(val.IsLocation());
 
         GenTree* indir = val.Node();
-        assert(indir->OperIs(GT_IND, GT_FIELD));
+        assert(indir->OperIs(GT_IND, GT_OBJ, GT_BLK, GT_FIELD));
 
-        if ((indir->OperIs(GT_IND) && indir->AsIndir()->IsVolatile()) ||
-            (indir->OperIs(GT_FIELD) && indir->AsField()->IsVolatile()))
+        if (val.Offset() > UINT16_MAX)
         {
+            // TODO-ADDR: We can't use LCL_FLD because the offset is too large but we should
+            // transform the tree into IND(ADD(LCL_VAR_ADDR, offset)) instead of leaving this
```

Comment:
Thanks, I have the habit of changing comments multiple time because I'm not happy with the phrasing and sometimes I mess things up. I think it can wait until a future PR, unless we want to restart the CI to deal with the ARM failure.